src/modules/assets/assets.subscriber.ts

```diff
@@ -86,9 +89,21 @@ export class AssetsEventSubscriber {
     const saved = await this.historyRepository.save(eventToSave);
     this.logger.debug(
       `${eventToSave.type} event for ${
-        eventToSave.assetId
+      eventToSave.assetId
       } handled: ${JSON.stringify(saved)}`,
     );
+
+    const msg = {
+      event: MessageType.AssetOffered,
+      asset: event.assetId,
+      owner: event.emittedBy,
+      offeredTo: event.relatedTo,
+      at: event.at
+    };
+    this.nats.connection.publish(
+      `${msg.offeredTo}.${NATS_EXCHANGE_TOPIC}`,
+      JSON.stringify(msg)
+    );
```

Comment:
It looks like this code is duplicated a few times. Maybe it makes sense to have function like `publishMessage(type, targetDID)`?